great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -109,7 +109,13 @@ def __init__(
                 "schema_name can only be used with table_name. Use temp_table_schema_name to provide a target schema for creating a temporary table."
             )
 
-        dialect: GESqlDialect = GESqlDialect(engine.dialect.name.lower())
+        dialect_name: str = engine.dialect.name.lower()
+
+        try:
+            dialect: GESqlDialect = GESqlDialect(dialect_name)
+        except ValueError:
+            dialect: GESqlDialect = GESqlDialect.UNSUPPORTED
```

Comment:
If you want to store the unsupported dialect name you could hang it off dialect as a quick hack. That is follow this with the line `dialect._unsupported_dialect = dialect_name`. You would want to do that below also. If we end up doing anything with this value, we should do something better as a followup.

Reply:
Another idea (not needed now)
Would be do do this logic within `__missing__` which gets called when the value isn't a member.
Which also mean we could get rid of this try: expect.

https://docs.python.org/3/library/enum.html#supported-sunder-names